uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -1176,6 +1186,30 @@ define([
       }
 
       /**
+       * Fire and event to start Uploading attachment and resolves it according to the result
+       *
+       * @returns {Promise}
+       */
+      function uploadAttachment () {
+        var deferred = $q.defer();
+
+        var successEvent = $rootScope.$on('uploadFiles: success', function () {
```

Comment:
i think this is overcomplicate. You can simply set the listeners in the `init` of the controller, rather than creating and destroying them on the fly